src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+const avatarInput = avatarModal.querySelector('#profile-avatar-input');
+
+// Delete Modal elements
+const deleteModal = document.querySelector('#delete-modal');
+const deleteForm = deleteModal.querySelector('.modal__form');
+const deleteModalCancelBtn = deleteModal.querySelector('#delete-cancel-btn');
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add('modal_opened');
+  document.addEventListener('keydown', handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_opened');
+  document.removeEventListener('keydown', handleEscapeKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // editModalSubmitBtn.textContent = 'Saving...';
+  // change text content to "Saving..."
+  const editModalSubmitBtn = evt.submitter;
+  editModalSubmitBtn.textContent = 'Saving...';
+
+  setButtonText(editModalSubmitBtn, true, "Save", "Saving...");
+  // disable the button
+  api.editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      // todo - call setButtonText instead
+      editModalSubmitBtn.textContent = "Save";
+    });
+}
+
+// todo - implement loading text for all other form submissions
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const addCardModalSubmitBtn = evt.submitter;
+  addCardModalSubmitBtn.textContent = 'Saving...';
+  setButtonText(addCardModalSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      console.log(data);
+
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+    }).catch(console.error)
+    .finally(() => {
+
+      addCardModalSubmitBtn.textContent = "Save";
+    });
+  disableButton(cardSubmitBtn, settings);
+  closeModal(cardModal);
+
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
+
+  avatarSubmitBtn.textContent = 'Saving...';
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  console.log(avatarInput.value);
+  // const formData = new FormData(avatarForm);
+  // const avatarUrl = formData.get('profile-avatar-input');
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      console.log(data);
+      // const avatarImageSrc = document.getElementById("profile-avatar");
+      avatarImageSrc.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      avatarSubmitBtn.textContent = "Save";
+    });
+  evt.target.reset();
+  disableButton(avatarSubmitBtn, settings);
```

Comment:
These 2 lines should be in `.then` in li ne 207


You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.